content/blog/2020-12-19-LLVM-IR-to-assembly.md

```diff
@@ -0,0 +1,77 @@
+The benchmark is primarily done by FFT and is implemented in SSE.   Thie code is compiled by three diffeent compliers: a unmodified version of GCC, unmodified version of clang, and a modified version of clang with only 75% of the original x86-64 registers.     
+The benchmark is done on a 4 core i9-9900K CPU with 16 GB RAM in a Ubuntu VM.  The results are reported cycle count with rdtsc and execution time is reported as wall-clock time.  The results are verified by comparing the results of different implementations.
+
+
+## Microbenchmark 
+
+### Process and Methology
+The microbenchmark preforms a 4x4 matrix multiplication.  A total of 4096 itnerations have been preformed, and the results are aggerate by the minimum function.  Due to the small amount of cycle it takes to compelte the opereation, using wall-clock time may introduced another source of error, therefore I have elected to report the results in CPU cycles.
+
+
+### Results
+Table 1, Microbenchmark of 4x4 matrix in CPU Cycles, 4096 trials, Minimum Number cycles required to complete the operation. 
+
+| Microbenchmark 4x4   matrix             | gcc                   | clang  | clang (0.75) |
+|-----------------------------------------|-----------------------|--------|--------------|
+| REF                                     | 227.76                | 214.98 | 214.8        |
+| SSE                                     | 144.06                | 124.48 | 124.33       |
+| AV_4mem                                 | 140.11                | 170.03 | 170.14       |
+| AVX_8                                   | 93.48                 | 98.99  | 95.75        |
+
+### Results and Analysis
+As this microbenchmark is designed to measure the maximum performance of the complied code, I have chosen to aggregate the result a min function. Preliminary analysis has shown that reducing the number of architectural registers has little or no effect on the performance on microbenchmarks.  This is as expected as microbenchmarks do not have high amounts register pressure as they often contain couple of register operands.  In this case, the complier backend implementation has more of an impact on the final performance then the modification on the number of x86-64 architectural registers.  
+
+
+## FFT benchmark:
+
+### Process and Methology
+The benchmark preforms a set of FET calculations.  A total of 6 interactions have been performed and the data was preformed statistical analysis.  Execution time is measured in wall-clock time.
+
+### Results
+Table 2, FET benchmark in wall-clock time,average number cycles required to complete the operation.  
```

Comment:
I'm still confused here. Is this cycles or seconds? It looks like cycles, but then why the mention of wall-clock time?